articlequality/extractors/tests/test_ptwiki.py

```diff
@@ -19,54 +19,149 @@ def __init__(self, title, namespace, revisions):
+                "|qualidade=5|importância=2}}"
+            ),
+            Revision(
+                7, Timestamp(7), "ggg",
+                "{{Marca de projeto|AB}}"
+            ),
+            Revision(
+                8, Timestamp(8), "hhh",
+                "{{Marca de projeto|AD|Biografias|4}}"
+            )
+        ]),
+        Page("Page with single revert", 1, [
+            Revision(
+                1, Timestamp(1), "aaa",
+                "{{Marca de projeto|2}}"
+            ),
+            Revision(
+                2, Timestamp(2), "bbb",
+                "{{Marca de projeto|3}}"
+            ),
+            Revision(
+                3, Timestamp(3), "ccc",
+                "{{Marca de projeto|4}}"
+            ),
+            Revision(
+                4, Timestamp(4), "aaa",
+                "{{Marca de projeto|2}}"  # Vandal messing up the template
+            ),
+            Revision(
+                5, Timestamp(5), "ccc",
+                "{{Marca de projeto|4}}"  # Patroller reverting vandal
+            )
+        ]),
+        Page("Page with overlaping reverts", 1, [
+            Revision(
+                1, Timestamp(1), "aaa",
+                "{{Marca de projeto|1}}"
+            ),
+            Revision(
+                2, Timestamp(2), "bbb",
+                "{{Marca de projeto|2}}"
+            ),
+            Revision(
+                3, Timestamp(3), "ccc",
+                "{{Marca de projeto|3}}"
+            ),
+            Revision(
+                4, Timestamp(4), "aaa",
+                "{{Marca de projeto|1}}"  # Vandal messing up the template
+            ),
+            Revision(
+                5, Timestamp(5), "ccc",
+                "{{Marca de projeto|3}}"  # Rollback
+            ),
+            Revision(
+                6, Timestamp(6), "bbb",
+                "{{Marca de projeto|2}}"  # Active editor reevaluates the page
+            ),
+            Revision(
+                7, Timestamp(7), "ddd",
+                "{{Marca de projeto|4}}"  # Later on, the page is improved
+            )
+        ]),
+        Page("Page with concentric reverts", 1, [
+            Revision(
+                1, Timestamp(1), "aaa",
+                "{{Marca de projeto|1}}"
+            ),
+            Revision(
+                2, Timestamp(2), "bbb",
+                "{{Marca de projeto|2}}"
+            ),
+            Revision(
+                3, Timestamp(3), "ccc",
+                "{{Marca de projeto|3}}"
+            ),
+            Revision(
+                4, Timestamp(4), "aaa",
+                "{{Marca de projeto|1}}"  # Vandal messing up the template
+            ),
+            Revision(
+                5, Timestamp(5), "ccc",
+                "{{Marca de projeto|3}}"  # Rollback
+            ),
+            Revision(
+                6, Timestamp(6), "ddd",
+                "{{Marca de projeto|2}}"  # Active editor reevaluates the page
```

Comment:
Supposedly, the sha1 of two revisions with the same content should be the same.
Therefore, shouldn't we change `{{Marca de projeto|2}}` to something else just to keep this implicit assumption valid, while keeping the same label?
E.g.: `{{Marca de projeto|2}}<!-- Reevaluated by John -->`

Reply:
{{done}}